### PR TITLE
support cleaning files by extensions

### PR DESCRIPTION
**Pull request by nguyenhiepvan**

### CODE REVIEW

**Comment by freekmurze** [comment]
Please make sure that all tests pass.

**Comment by nguyenhiepvan** [comment]
> Please make sure that all tests pass.

done

File: src/DirectoryCleaner.php [hunk 1]
```diff
@@ -58,7 +69,13 @@ public function deleteFilesOlderThanMinutes($amountOfFilesDeleted = 0, $director
                 return $this->policy()->shouldDelete($file);
             })
             ->each(function ($file) {
-                $this->filesystem->delete($file);
+                if ($this->extensions){
+                    if (in_array($file->getExtension(),$this->extensions)){
+                        $this->filesystem->delete($file);
+                    }
+                }else{
```
Comment (freekmurze) [inline src/DirectoryCleaner.php:76]: Avoid using `else`. Maybe you could add a protected function `shouldDelete` and use early returns there.

File: src/DirectoryCleaner.php [hunk 2]
```diff
@@ -30,6 +33,14 @@ public function setDirectory(string $directory)
         return $this;
     }
 
+    public function setExtensions(string $extensions)
+    {
+        if ($extensions && $extensions != "*"){
```
Comment (freekmurze) [inline src/DirectoryCleaner.php:38]: Please use strict comparisons.

Use an early return if you don't want to do anything on `*`

**Review by freekmurze — CHANGES_REQUESTED** [review]
Please add tests that prove that the new functionality is working correctly.

Also add examples of the new functionality to the docs.

**Comment by freekmurze** [comment]
To keep the package as simple as it can be, I've decided not to merge this in.

I don't need the changes myself and I haven't received request for this, from other users.

But your work is not lost: feel free to create a fork and apply your changes there.

Thank you!

**Pull request closed without merging**